app/renderer/util.js

```diff
@@ -139,7 +137,25 @@ function getUniqueXPath(doc, domNode, attrs) {
   let uniqueXpath, semiUniqueXpath;
   const tagForXpath = domNode.tagName || '*';
   const isPairs = attrs.length > 0 && _.isArray(attrs[0]);
+  const isNodeName = attrs.length === 0;
+
+  // If we're looking for a unique //<nodetype>, return it only if it's actually unique. No
+  // semi-uniqueness here!
+  if (isNodeName) {
+    let xpath = `//${domNode.tagName}`;
+    const [isUnique] = determineXpathUniqueness(xpath, doc, domNode);
+    // even if this node name is unique, if it's the root node, we don't want to refer to it using
+    // '//' but rather '/'
+    if (!domNode?.parentNode?.tagName) {
+      xpath = `/${domNode.tagName}`;
+    }
```

Comment:
I'm not sure how useful this condition is. From the apps I checked, Android always has two `android.widget.FrameLayout` as the roots, so `isUnique` will be false anyway; and iOS has `XCUIElementTypeApplication` with filled out `name` and `label` fields, so the xpath generator will rely on those, and not even reach this base case. So currently I haven't found a usecase for this.

Also, CI is failing here because we can't use optional chaining (Parcel v1 limitation). It probably needs to be something like `if (!(domNode.parentNode && domNode.parentNode.tagName))`.

Finally, if we really must keep the condition, it should probably be moved inside the `isUnique` block on the next line.